neurogym/wrappers/monitor.py

```diff
@@ -1,36 +1,53 @@
+import tempfile
 from pathlib import Path
 
+import matplotlib.pyplot as plt
 import numpy as np
 from gymnasium import Wrapper
 
 from neurogym.utils.plotting import fig_
 
 
 class Monitor(Wrapper):
-    """Monitor task.
+    """Monitor wrapper for NeuroGym environments.
 
-    Saves relevant behavioral information: rewards, actions, observations, new trial, ground truth.
+    This class wraps NeuroGym environments to monitor and collect behavioral data
+    during training and evaluation. It saves relevant behavioral information such as
+    rewards, actions, observations, new trial flags, and ground truth.
+
+    DATA COLLECTION BEHAVIOR:
+    - The Monitor records data points ONLY at the end of trials (when info["new_trial"]=True).
+    - Each data point represents one complete behavioral trial, not individual timesteps.
+    - For NeuroGym tasks, this typically occurs when the agent makes a decision in the
+      decision period, or when the trial is aborted.
+    - Data is saved to disk at regular intervals (e.g., every 1000 trials) and internal
+      data containers are reset after each save.
 
     Args:
-        folder: Folder where the data will be saved. (def: None, str)
-            sv_per and sv_stp: Data will be saved every sv_per sv_stp's.
-            (def: 100000, int)
-        verbose: Whether to print information about average reward and number
-            of trials. (def: False, bool)
-        sv_fig: Whether to save a figure of the experiment structure. If True,
-            a figure will be updated every sv_per. (def: False, bool)
-        num_stps_sv_fig: Number of trial steps to include in the figure.
-            (def: 100, int)
+        env: The NeuroGym environment to wrap
+        folder: Path to folder where data will be saved. If None, uses a temporary directory.
+        sv_per: How often to save data (in trials or timesteps, depending on sv_stp)
+        sv_stp: Unit for sv_per, either "trial" or "timestep"
+        verbose: If True, prints information about average reward and number of trials
+        sv_fig: If True, saves figures visualizing the experiment structure
+        num_stps_sv_fig: Number of trial steps to include in each visualization figure
+        name: Optional name suffix for saved files
+        fig_type: File format for saved figures (e.g., 'png', 'svg', 'pdf')
+        step_fn: Optional custom step function to use instead of env.step
+
+    Attributes:
+        data: Dictionary containing behavioral data (actions, rewards, etc.)
+        num_tr: Number of trials completed
+        t: Number of timesteps completed (when sv_stp="timestep")
     """
 
-    metadata: dict[str, str | None] = {  # noqa: RUF012
+    metadata = {  # noqa: RUF012
         "description": (
             "Saves relevant behavioral information: rewards, actions, observations, new trial, ground truth."
         ),
         "paper_link": None,
         "paper_name": None,
     }
```

Comment:
why is this metadata needed at all? Isn't it basically what the docstring is?

Reply:
They're in all envs, it's a class variable which is then used in `TrialEnv` in the `__str__` method

Reply:
This isn't an env, though and I don't think that wrapper metadata is used anywhere. 
I do see that other wrappers have the metadata as well. In Gymnasium, usually the envs all have this metadata variable, but the wrappers don't. I guess this is also something that can be a separate refactoring, to remove metadata variables from the wrappers.